CRM/Contribute/BAO/ContributionRecur.php

```diff
@@ -151,6 +157,28 @@ public static function checkDuplicate($params, &$duplicates) {
     }
     return $result;
   }
+  
+  /**
+   * @param bool $test
+   * @param $mode
+   *
+   * @return array|null
+   */
+  public static function getBackOfficePaymentProcessors($test = FALSE, $mode = 'live') {
```

Comment:
Take a look at how other forms are finding backoffice processors. Basically all the backoffice forms call

$this->assignPaymentRelatedVariables();

Which among other things adds

      $this->_paymentProcessors = $this->getValidProcessors();

to the form.

I've been working pretty hard at removing as many references to 'billing_mode' & 'payment_type' from the code as humanly possible. 

I would hope this function can be removed in favour of relying on the mechanisms mentioned above.